new_structure/views/auth.py

```
@@ -67,24 +88,35 @@ def admin_login():
+                'teacher_id': int(os.getenv('TEACHER_TEACHER_ID', '4')),
+                'role': 'headteacher'
+            }
+        }
+
+        if username in valid_credentials and password == valid_credentials[username]['password']:
+            print(f"🔍 Authentication successful via bypass mechanism from IP: {client_ip}")
+            session['teacher_id'] = valid_credentials[username]['teacher_id']
+            session['username'] = username
+            session['role'] = valid_credentials[username]['role']
             session.permanent = True
+            print(f"🔍 Session set, redirecting to proper headteacher dashboard...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)